frida_mode/src/persistent/persistent_arm64.c

```diff
@@ -105,6 +105,7 @@ static void instrument_persitent_save_regs(GumArm64Writer *  cw,
                                           offsetof(persistent_ctx_t, rflags));
 
   /* Q */
+  #if defined(__arm64__)
```

Comment:
My bad, its not needed. I had compile errors coming from that in my first setup. Now it only breaks by frida 15.1.25.

Reply:
![grafik](https://user-images.githubusercontent.com/15239745/175278885-6b5a2c08-9b5d-4987-8113-d3cbfb293763.png)

Reply:
I think Ole added more support for NEON/SIMD registers recently which maybe have changed the structure slightly....
https://github.com/frida/frida-gum/blob/eb4faac39fa2db6d221d329ee50ca2c8059543f1/gum/gumdefs.h#L276

Reply:
Okey, so I might go against the latest version 15.1.27 then.
The variable naming in the x86-backend also changed. After changing that I could run some tests on my x64 successfully.

My problem is just that I dont have figured out how to use persistent mode on android yet. The addr binary is not working. I will look into the test examples and addr source and try to figure out my alternative.

Reply:
Yeah. Bumping to the latest certainly shouldn't do any harm. The `addr` binary is just used to work out the default load address for PIE executables when ASLR is disabled.

Rather than relying on an...
```
echo 0 > /proc/sys/kernel/randomize_va_space
```
FRIDA mode instead uses the `personality` API with `ADDR_NO_RANDOMIZE`. This takes effect from the next `execve`, so the process then `execve`'s itself to cause it to be loaded at a fixed address (without requiring any system-wide configuration). Hence the need to hook `main` or similar to access the arguments and environment so that they can be passed to the `execve` function to ensure that the new process keeps all of it's options.

Reply:
Yeah. Bumping to the latest certainly shouldn't do any harm. The `addr` binary is just used to work out the default load address for PIE executables when ASLR is disabled.

Rather than relying on an...
```
echo 0 > /proc/sys/kernel/randomize_va_space
```
FRIDA mode instead uses the `personality` API with `ADDR_NO_RANDOMIZE`. This takes effect from the next `execve`, so the process then `execve`'s itself to cause it to be loaded at a fixed address (without requiring any system-wide configuration). Hence the need to hook `main` or similar to access the arguments and environment so that they can be passed to the `execve` function to ensure that the new process keeps all of it's options.

`addr` is used by the makefile for various tests to help in calculating the address of symbols in the target application. `addr` provides the base address, and [get_symbol_addr.sh](https://github.com/AFLplusplus/AFLplusplus/blob/stable/frida_mode/util/get_symbol_addr.sh) provides the offset (by means of grepping the output of `nm`). This allows variables such as `AFL_FRIDA_PERSISTENT_ADDR` to be set appropriately when testing with persistent mode.

Of course, using the [scripting](https://github.com/AFLplusplus/AFLplusplus/blob/stable/frida_mode/Scripting.md) mode avoids all this since you can use FRIDA's symbol engine to find symbols of interest and use the API exposed by AFL++ FRIDA mode to configure these settings.